corehq/apps/locations/middleware.py

```diff
@@ -42,6 +42,11 @@ def process_view(self, request, view_fn, view_args, view_kwargs):
         else:
             request.can_access_all_locations = False
             if not is_location_safe(view_fn, view_args, view_kwargs):
+                # is_location_safe() is returning False because the wrapped (wrapped, wrapped (wrapped?)) view_fn
+                # no longer has is_location_safe attribute set on it.
+                # TODO: Find where it's being dropped, and set it on the wrapper function from the wrapped function
+                #       ^^^ For *every* code path? See corehq.apps.domain.decorators. This doesn't seem like such
+                #       a good idea any more.
```

Comment:
i am not sure its getting dropped, i think it hasnt been set yet because the view executes (with the wrapping decorator) after the middleware finishes.